migration/migration.go

```diff
@@ -429,3 +432,100 @@ func loadFields(ctx context.Context, wit *workitem.WorkItemType, into workitem.F
 
 	return nil
 }
+
+// migrates work item fields stored in a JSON document by converting the current 'system.description'
+// string element into an object with 2 fields: 'raw_value' and 'markup_type'  (related to issue #477)
+func migrateWorkItemDescriptionsToMarkup() fn {
+	return func(tx *sql.Tx) error {
+		// skip if the db version is >= 17, which is when the description has been migrated to the 'MarkupType'
+		versionRow := tx.QueryRow("SELECT max(version) as current FROM version")
+		var currentVersion int64 = -1
+		if err := versionRow.Scan(&currentVersion); err != nil {
+			return fmt.Errorf("Failed to scan max version in table \"version\": %s\n", err)
+		}
+		if currentVersion >= 17 {
+			return nil
+		}
+
+		totalRows, err := tx.Query("SELECT count(*) FROM work_items")
+		if err != nil {
+			return fmt.Errorf("Failed to count work items: %s\n", err)
+		}
+		var totalCount int
+		totalRows.Next()
+		totalRows.Scan(&totalCount)
+		fmt.Println("Migrating", totalCount, "work item description(s) to 'markup' structure...")
+		totalRows.Close()
+
+		rows, err := tx.Query("SELECT id, fields FROM work_items ORDER BY id")
+		if err != nil {
+			return fmt.Errorf("Failed to select work items: %s\n", err)
+		}
+		// convert 'fields' for each row, then update in db
+		allUpdatedFields := make(map[int]string)
+		for rows.Next() {
+			var id int
+			var fields string
+			err = rows.Scan(&id, &fields)
+			var dat map[string]interface{}
+			if err := json.Unmarshal([]byte(fields), &dat); err != nil {
+				panic(err)
+			}
+			fmt.Println("Found row with id", id, "and description:", dat[workitem.SystemDescription])
+			descriptionContent := getDescriptionContent(dat)
+			if descriptionContent == nil {
+				fmt.Println("Skipping description update for work item with id", id, "as there is no description")
+				continue
+			}
+			descriptionMarkup := getDescriptionMarkup(dat)
+			description := workitem.MarkupContent{
+				Content: *descriptionContent,
+				Markup:  *descriptionMarkup,
+			}
+			dat[workitem.SystemDescription] = description
+			updatedFields, err := json.Marshal(dat)
+			if err != nil {
+				panic(err)
+			}
+			allUpdatedFields[id] = string(updatedFields)
+		}
+		rows.Close()
+		// prepared statement
+		updateFieldsStmt, err := tx.Prepare("UPDATE work_items SET fields=$1 WHERE id=$2")
+		if err != nil {
+			return fmt.Errorf("Failed to prepare statement to update work items: %s\n", err)
+		}
+		for id, updatedFields := range allUpdatedFields {
+			fmt.Println("Updated fields:", string(updatedFields))
+
+			if _, err := tx.Stmt(updateFieldsStmt).Exec(updatedFields, id); err != nil {
+				return fmt.Errorf("Failed to update work item with id %d with fields value \"%s\" : %s", id, string(updatedFields), err.Error())
```

Comment:
Import github.com/pkg/errors and return

```go
return errors.Wrapf(err, "Failed to update work item with id %d with fields value \"%s\"", id, string(updatedFields))
```

See https://godoc.org/github.com/pkg/errors#Wrapf for an example and why we want to do that at least for new errors. @aslakknutsen please stop me if you don't find this useful.